hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1395,11 +1406,29 @@ bool MDMParser::checkNetStatus(NetStatus* status /*= NULL*/)
         if (RESP_OK != waitFinalResp(_cbCOPS, &_net, COPS_TIMEOUT)) {
             goto failure;
         }
-        // get the signal strength indication
-        sendFormated("AT+CSQ\r\n");
-        if (RESP_OK != waitFinalResp(_cbCSQ, &_net, CSQ_TIMEOUT)) {
-            goto failure;
+        // AT command used to collect signal stregnth is different for R410M radio
+        if (_dev.dev == DEV_SARA_R410) {
+            sendFormated("AT+UCGED=5\r\n");
+            if (RESP_OK != waitFinalResp(nullptr, nullptr, UCGED_TIMEOUT)) {
+                goto failure;
+            }
+            // Reset the ucged parsing flags before checking for UCGED command
+            ucged_parse_rsrp_ok = false;
+            ucged_parse_rsrq_ok = false;
+            sendFormated("AT+UCGED?\r\n");
+            if ((RESP_OK != waitFinalResp(_cbUCGED, &_net, UCGED_TIMEOUT))
+                            || (!ucged_parse_rsrp_ok || !ucged_parse_rsrq_ok)) { // rsrp or rsrq not parsed correctly in ucged
+                goto failure;
```

Comment:
I think it's ok if ucged_parse_rsrp_ok and ucged_parse_rsrq_ok are false... we don't have to `goto failure;` in that case.  These should only be used to invalidate the rsrp and rsrq values.